src/setup/config.rs

```diff
@@ -18,6 +23,56 @@ pub struct NodeConfig {
     pub log_to_stdout: bool,
     /// The path of the cache directory of the node.
     pub path: PathBuf,
+    /// The network socket address of the node.
+    pub local_addr: Option<SocketAddr>,
+    /// The REST API socket address of the node.
+    pub rest_api_addr: Option<SocketAddr>,
+}
+
+impl NodeConfig {
+    /// Continuously try to read a string from a file.
+    async fn try_read_to_string(file_path: &Path) -> String {
+        loop {
+            match fs::read_to_string(&file_path) {
```

Comment:
since we're already doing `async`, why not https://docs.rs/tokio/latest/tokio/fs/fn.read_to_string.html?